src/biogeochem/Fan2CTSMMod.F90

```diff
@@ -0,0 +1,1059 @@
+    use clm_varcon, only : ispval
+    use landunit_varcon,      only:  istsoil, istcrop
+    use abortutils     , only : endrun
+    use LandunitType   , only: lun
+    use GridcellType   , only: grc
+    use clm_varctl     , only : iulog
+
+    implicit none
+    type(bounds_type), intent(in)    :: bounds
+    type(temperature_type) , intent(in) :: temperature_inst
+    type(frictionvel_type) , intent(in) :: frictionvel_inst
+    real(r8), intent(in) :: dt
+    
+    ! N excreted in manure, gN/m2:
+    real(r8), intent(in) :: ndep_sgrz_grc(bounds%begg:bounds%endg) ! seasonally grazing animals
+    real(r8), intent(in) :: ndep_ngrz_grc(bounds%begg:bounds%endg) ! non-grazing animals
+    ! N, TAN spread on grasslands, gN/m2/s:
+    real(r8), intent(inout) :: n_manure_spread_col(bounds%begc:bounds%endc) 
+    real(r8), intent(out) :: tan_manure_spread_col(bounds%begc:bounds%endc) ! output, calculated from the above and stored manure
+    ! N excreted by animals allocated to mixed production systems temporarily grazing on grasslands:
+    real(r8), intent(inout) :: n_manure_graze_col(bounds%begc:bounds%endc)
+    ! N excreted by animals in mixed systems, total
+    real(r8), intent(out) :: n_manure_mixed_col(bounds%begc:bounds%endc)
+    ! NH3 emission fluxes from manure storage and housings, gN/m2/s
+    real(r8), intent(out) :: nh3_flux_stores(bounds%begc:bounds%endc), nh3_flux_barns(bounds%begc:bounds%endc)
+    ! total nitrogen flux transferred out of a crop column (manure spreading + temporary grazing)
+    real(r8), intent(out) :: man_n_transf(bounds%begc:bounds%endc)
+    ! Total nitrogen excreted in barns
+    real(r8), intent(out) :: man_n_barns(bounds%begc:bounds%endc)
+    ! fraction of manure excreted when grazing
+    real(r8), intent(out) :: grz_fract(bounds%begc:bounds%endc)
+    ! TAN fraction in excreted N
+    real(r8), intent(in) :: tan_fract_excr
+    integer                  , intent(in)    :: num_soilc       ! number of soil columns in filter
+    integer                  , intent(in)    :: filter_soilc(:) ! filter for soil columns
+
+    integer :: begg, endg, g, l, c, il, counter, col_grass, status, p
+    real(r8) :: flux_avail_rum, flux_avail_mg, flux_grazing, invscale
+    real(r8) :: tempr_ave, windspeed_ave ! windspeed and temperature averaged over agricultural patches
+    real(r8) :: tempr_barns, tempr_stores, vent_barns, flux_grass_crop, tempr_min_10day, &
+         flux_grass_graze, flux_grass_spread, flux_grass_spread_tan, flux_grass_crop_tan
+    real(r8) :: cumflux, totalinput, total_to_store, total_to_store_tan
+    ! dimensions are (type of flux, ruminant/other)
+    real(r8) :: fluxes_nitr(num_fluxes,2), fluxes_tan(num_fluxes,2) 
+    ! The fraction of manure applied continuously on grasslands (if present in the gridcell)
+    real(r8), parameter :: kg_to_g = 1e3_r8
+    logical :: is_grass
+    
+    begg = bounds%begg; endg = bounds%endg
+    nh3_flux_stores(bounds%begc:bounds%endc) = 0_r8
```

Comment:
Also, here and elsewhere: a decimal is needed to make 0 a real value. Otherwise, the _r8 doesn't do at all what you'd expect: Here, the 0 here is an integer 0 with some funny kind, NOT a double-precision real as is intended.